java/main/com/google/jglome/Glome.java

```diff
@@ -0,0 +1,284 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.jglome;
+
+import static java.lang.System.arraycopy;
+
+import com.google.crypto.tink.subtle.X25519;
+import java.nio.ByteBuffer;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Encapsulates the logic of GLOME protocol.
+ */
+public final class Glome {
+
+  private byte[] peerKey;
+  private Mac userMacKey;
+  private Mac peerMacKey;
+  private int minPeerTagLength;
+
+  private final byte[] userPublicKey;
+  private final byte[] userPrivateKey;
+
+  public static final int MIN_CNT_VALUE = 0;
+  public static final int MAX_CNT_VALUE = 255;
+  public static final int MAX_TAG_LENGTH = 32;
+  public static final int MIN_TAG_LENGTH = 1;
+
+  /**
+   * Builder class for Glome object.
+   */
+  public static class GlomeBuilder {
+
+    private final byte[] peerKey;
+    private final int minPeerTagLength;
+    private byte[] userPublicKey;
+    private byte[] userPrivateKey;
+    private Mac userMacKey;
```

Comment:
What's with the `Key` suffix here? I can't find a reason why it is just not `userMac` / `peerMac`.